dialoptions.go

```diff
@@ -96,6 +97,16 @@ type EmptyDialOption struct{}
 
 func (EmptyDialOption) apply(*dialOptions) {}
 
+type disableGlobalDialOptions struct{}
+
+func (disableGlobalDialOptions) apply(*dialOptions) {}
+
+// newDisableGlobalDialOptions returns a DialOption that prevents the ClientConn
+// from applying the global DialOptions (set via AddGlobalDialOptions).
+func newDisableGlobalDialOptions() DialOption {
```

Comment:
Nit/bikeshed: `s/new//`?  Then you'd have to rename `disableGlobalDialOptions` to `disableGlobalDialOptionsOption`.  Maybe this is fine then.

Reply:
Ok, will leave as such.